app/services/salesforce_subscriber.rb

```diff
@@ -13,27 +13,29 @@ def initialize
   end
 
   def create_contact_push_topic
-    delete_contact_topics
-    contact_topic = @client.create!('PushTopic',
-                                   ApiVersion: '51.0',
-                                   Name: CONTACT_PUSH_TOPIC_NAME,
-                                   Description: 'all contact records',
-                                   NotifyForOperationCreate: 'true',
-                                   NotifyForOperationUpdate: 'true',
-                                   NotifyForFields: 'Referenced',
-                                   Query: 'select Id, Email, Faculty_Verified__c from Contact')
+    topic = PushTopic.where(topic_name: CONTACT_PUSH_TOPIC_NAME).first
 
-    if contact_topic.present? && contact_topic.is_a?(String)
-      PushTopic.create(topic_salesforce_id: contact_topic, topic_name: CONTACT_PUSH_TOPIC_NAME)
-      Rails.logger.debug('Contact Push Topic Id: ' + contact_topic)
-    else
-      Rails.logger.error('failed to create push topic: '+ CONTACT_PUSH_TOPIC_NAME)
-      Sentry.capture_message('failed to create push topic: '+ CONTACT_PUSH_TOPIC_NAME)
-      raise
+    unless topic
+      contact_topic = @client.create!('PushTopic',
+                                      ApiVersion: '51.0',
+                                      Name: CONTACT_PUSH_TOPIC_NAME,
+                                      Description: 'all contact records',
+                                      NotifyForOperationCreate: 'true',
+                                      NotifyForOperationUpdate: 'true',
+                                      NotifyForFields: 'Referenced',
+                                      Query: 'select Id, Email, Email_alt__c, Faculty_Verified__c, Adoption_Status__c, Grant_Tutor_Access__c from Contact')
+
+      if contact_topic.present? && contact_topic.is_a?(String)
+        PushTopic.create(topic_salesforce_id: contact_topic, topic_name: CONTACT_PUSH_TOPIC_NAME)
+        warn('Contact Push Topic Id: ' + contact_topic)
+      else
+        Rails.logger.error('failed to create push topic: ' + CONTACT_PUSH_TOPIC_NAME)
+        Sentry.capture_message('failed to create push topic: ' + CONTACT_PUSH_TOPIC_NAME)
+        raise
+      end
     end
   rescue Restforce::ErrorCode::DuplicateValue
-    delete_contact_topics
-    create_contact_push_topic
+    Rails.logger.debug('Push topic duplicate found.')
```

Comment:
Might wanna call `retry` here (I think) in case the topic got created in between the 2 calls. And it's a good idea to have a max_retry...

Reply:
Didn't know about the `retry` command - cool! Added.